src/key_packages/mod.rs

```diff
@@ -195,10 +194,10 @@ impl KeyPackage {
         &self.extensions
     }
 
-    /// Sign the key package.
-    pub(crate) fn sign(&mut self, ciphersuite: &Ciphersuite, signature_key: &SignaturePrivateKey) {
+    /// Populate the `signature` field using the `credential_bundle`.
+    pub(crate) fn sign_self(&mut self, credential_bundle: &CredentialBundle) {
```

Comment:
Why `sign_self`? The `&mut self` should give that away 🙂

Reply:
Because if we call it `sign`, then the function is hidden behind the implementation of `signable`, at least that's how I understand it. Is there a better way of doing this?

Reply:
Ah, this implements signable? Maybe we can call this something like `re_sign` (not resign 😆 ) if it's only used to re-sign. You can still call it even if signable is implemented, but having clashing names is never nice. So pick the name you like best or keep the `sign_self`.

Reply:
I think what happens is that the signature field is initialized as empty and then filled with this signature. So it's not (just) for re-signing. Since `Signable` is used nowhere else, I've removed it for now and renamed it to `sign`. I think `Signable` needs to be reworked anyway (see issue #160).